src/osg/State.cpp

```diff
@@ -1257,7 +1257,6 @@ namespace State_Utils
                 std::string::size_type start_var = pos;
                 ++pos;
                 pos = str.find_first_not_of("ABCDEFGHIJKLMNOPQRTSUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_", pos);
-                std::string var_str;
```

Comment:
Yay!  A bona fida bug, testing for this fix led me to find other bugs elsewhere in this function so I've ended up fixing two bugs.